modules/gdscript/gdscript_byte_codegen.cpp

```diff
@@ -585,8 +585,8 @@ void GDScriptByteCodeGenerator::write_unary_operator(const Address &p_target, Va
 }
 
 void GDScriptByteCodeGenerator::write_binary_operator(const Address &p_target, Variant::Operator p_operator, const Address &p_left_operand, const Address &p_right_operand) {
-	// Avoid validated evaluator for modulo and division when operands are int, since there's no check for division by zero.
-	if (HAS_BUILTIN_TYPE(p_left_operand) && HAS_BUILTIN_TYPE(p_right_operand) && ((p_operator != Variant::OP_DIVIDE && p_operator != Variant::OP_MODULE) || p_left_operand.type.builtin_type != Variant::INT || p_right_operand.type.builtin_type != Variant::INT)) {
+	// Avoid validated evaluator for modulo and division when operands are int or integer vector, since there's no check for division by zero.
+	if (HAS_BUILTIN_TYPE(p_left_operand) && HAS_BUILTIN_TYPE(p_right_operand) && ((p_operator != Variant::OP_DIVIDE && p_operator != Variant::OP_MODULE) || (!(p_left_operand.type.builtin_type == Variant::INT && p_right_operand.type.builtin_type == Variant::INT) && !(p_left_operand.type.builtin_type == Variant::VECTOR2I && (p_right_operand.type.builtin_type == Variant::VECTOR2I || p_right_operand.type.builtin_type == Variant::INT)) && !(p_left_operand.type.builtin_type == Variant::VECTOR3I && (p_right_operand.type.builtin_type == Variant::VECTOR3I || p_right_operand.type.builtin_type == Variant::INT)) && !(p_left_operand.type.builtin_type == Variant::VECTOR4I && (p_right_operand.type.builtin_type == Variant::VECTOR4I || p_right_operand.type.builtin_type == Variant::INT))))) {
```

Comment:
1. Could you please add tests for this?
2. Could you turn this into a `switch` for better readability?

Reply:
1. Not sure if I did the test right, can you check?
2. Fixed